### ISSUE TITLE
After starting and pressing R, the pause is switched

### ISSUE THREAD

**Issue by JackoKu** [till213/SkyDolly#124]
After starting and pressing R, the pause is switched.

I want to use VoiceAttack to start the program and then use R to record.
However, after starting the program and entering R, the program goes to "Pause".

It only works if I deactivate Pause and then press or leave R pressed.
So the error is that the program immediately goes to pause after starting and pressing R.

**Comment by till213** [comment]
Also refer to:

https://github.com/till213/SkyDolly/issues/106

**Comment by JackoKu** [comment]
Sorry, I keep having problems with the operation of Github.
As a normal user, I don't find Github intuitive.
That's why I probably inserted my comment in the wrong place and am repeating it here:

Because I was concentrating on the problem yesterday, I forgot to tell you what an excellent program you have written.
To come up with the idea of not only making an mp4 recording, but to make the playback in the simulator directly and thus see any perspective afterwards is ingenious in itself.
I am currently in the process of working out the many possibilities of the program.
One more question:
Can I somehow also manage an export, e.g. to mp4, direect from the SkyDolly recording?
Because if I have to start the MSFS, SkyDolly and OBS (to record SkyDolly recordings) in order to create a nice movie afterwards, this is very hard on the performance of the PC, which means that there are stutter at least at the start of the scene.

About the problem yesterday.
After I restarted the PC and MSFS for certain reasons, the problem no longer occurred.

**Comment by till213** [comment]
Hello,

First of all, thank you so much for your encouraging words and for using Sky Dolly. And second, I apologize for the long delay of my reply!

"Can I somehow also manage an export, e.g. to mp4, direect from the SkyDolly recording?"

No, this is not implemented in Sky Dolly. And to be quite honest, I am very reluctant in considering such a "screen / video capture" feature.

While in the meantime this should be possible with the underlying "framework" (the Qt toolkit specifically: https://www.qt.io/) to capture a separate window _content_ (even or especially if that window to be captured is - partially or fully - occluded by other windows) other tools such as the mentioned OBS (https://obsproject.com/) provide much much more control about the encoding settings such as the desired bitrate and encoder (h264, h265, ...) - also for the audio part.

Yes, you probably need quite a powerful PC to have a screen capture / encoder software and MSFS run at the same time (they typically compete for the GPU, as OBS is actually able to use the GPU as encoder - depending on the GPU). But the same would be true if Sky Dolly would make an attempt to capture the screen and encode it at the same time (note that Sky Dolly itself should be quite light on your CPU and RAM resources: sending a couple of dozens values (times the replay rate, like 30 to 60 Hz) and doing some floating point multiplications ("interpolation") in between is not very stressful for a modern CPU.

But who knows: since I thought about doing screen captures (as JPEG) for the Location module I might try how a - hypothetical ;) - video encoder would do. As a "simple solution", without all the bells and whizzles that OBS is providing. But no promises made here, as I really try to keep Sky Dolly lean and clean, from a usability point of view.

But hey: Sky Dolly already has a "modular architecture", so such a video recorder would of course be implemented as a "plugin" (like all the other plugins, including the Logbook and Formation module). And if someone would not like that module it could simply be removed from the plugin folder (or perhaps better: a "plugin manager" with which a user could de/activate certain plugins, without having to delete (move) them from the plugin folder).

If you are really interested in such a video encoder within Sky Dolly then I suggest to create a new issue here on github: it is good practice to keep different issues separate anyway ;)

**Comment by JackoKu** [comment]
So, you are absolutely right with your arguments.

I actually use OBS in parallel with SkyDolly, as I have a GPU RTX 4090 and a CPU i913900K. This also works without any major problems. Only sometimes (but rarely) the picture freezes for a few seconds. But you can cut that out later with a video editing programme.
The disadvantage is that with OBS I can of course no longer choose where and what I want to see (in the aircraft, outside the aircraft, tail unit or whatever).

That's why I'm currently recording with SkyDolly first.
Here I can choose which view I want during the replay, regardless of where I was when SkyDolly was recording in real time.
However, it is annoying that the flight plan in the ND display wobbles back and forth (see other ticket).

So, your idea with plugins is probably the best.

In fact, you should mainly make sure that you lose little or no fps with SkyDolly. That should be the main goal regardless of other ideas and gimmicks.
